sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc

```diff
@@ -94,514 +101,761 @@ implementation supports.
 
-|`amd_gpu_gfx908`
+a|
+[source]
+----
+amd_gpu_gfx908
+----
 |-
 |AMD CDNA 1 architecture.
 
-|`amd_gpu_gfx90a`
+a|
+[source]
+----
+amd_gpu_gfx909
+----
+|-
+|AMD GCN 5.0 architecture.
+
+a|
+[source]
+----
+amd_gpu_gfx90a
+----
 |-
 |AMD CDNA 2 architecture.
 
-|`amd_gpu_gfx940` to `amd_gpu_gfx942`
+a|
+[source]
+----
+amd_gpu_gfx90c
+----
+|-
+|AMD GCN 5.1 architecture.
+
+a|
+[source]
+----
+amd_gpu_gfx940
+amd_gpu_gfx941
+amd_gpu_gfx942
+----
 |-
 |AMD CDNA 3 architecture.
 
-|`amd_gpu_gfx1010` to `amd_gpu_gfx1013`
+a|
+[source]
+----
+amd_gpu_gfx1010
+amd_gpu_gfx1011
+amd_gpu_gfx1012
+amd_gpu_gfx1013
+----
 |-
 |AMD RDNA 1 architecture.
 
-|`amd_gpu_gfx1030` to `amd_gpu_gfx1036`
+a|
+[source]
+----
+amd_gpu_gfx1030
+amd_gpu_gfx1031
+amd_gpu_gfx1032
+amd_gpu_gfx1033
+amd_gpu_gfx1034
+amd_gpu_gfx1035
+amd_gpu_gfx1036
+----
 |-
 |AMD RDNA 2 architecture.
 
-|`amd_gpu_gfx1100` to `amd_gpu_gfx1103`
+a|
+[source]
+----
+amd_gpu_gfx1100
+amd_gpu_gfx1101
+amd_gpu_gfx1102
+amd_gpu_gfx1103
+----
 |-
 |AMD RDNA 3 architecture.
 
-|`amd_gpu_gfx1150` and `amd_gpu_gfx1151`
+a|
+[source]
+----
+amd_gpu_gfx1150
+amd_gpu_gfx1151
+----
 |-
 |AMD RDNA 3.5 architecture.
 
-|`amd_gpu_gfx1200` and `amd_gpu_gfx1201`
+a|
+[source]
+----
+amd_gpu_gfx1200
+amd_gpu_gfx1201
+----
 |-
 |AMD RDNA 4 architecture.
-
 |===
 
-[NOTE]
-====
-* An "alias" enumerator is generally added for new devices only after hardware
-has finalized and the exact version is known.
-* For NVIDIA GPUs, the architecture enumerator corresponds to the
-https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities[compute capability]
-of the device, and ext_oneapi_architecture_is can be used similarly to the \\__CUDA_ARCH__ macro in CUDA.
-====
+[_Note:_ An "alias" enumerator is generally added for new Intel GPU devices
+only after hardware has finalized and the exact version is known.
+_{endnote}_]
 
+[_Note:_
+For NVIDIA GPUs, the architecture enumerator corresponds to the
+https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities[compute capability]
+of the device, and `if_architecture_is` can be used similarly to the
+`+__CUDA_ARCH__+` macro in CUDA.
+_{endnote}_]
 
 === New `if_architecture_is` free function
 
-This extension adds one new free function which may be called from device
-code.  This function is not available in host code.
+This extension adds the following new free function which may be called from
+device code.
 
-```
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
 namespace sycl::ext::oneapi::experimental {
 
 template<architecture ...Archs, typename T>
 /* unspecified */ if_architecture_is(T fn);
 
 } // namespace sycl::ext::oneapi::experimental
-```
+----
+!====
 
-This function operates exactly like `if_device_has` from the
+_Constraints:_ The type `T` must be a {cpp} `Callable` type which is invocable
+with an empty parameter list.
+
+_Preconditions:_ This function must not be called from host code.
```

Comment:
Nit: I don't know if we have a precedent here, but I usually don't write these as negated conditions.

```suggestion
_Preconditions:_ This function must be called from device code.
```